src/main/java/io/demo/bank/service/VisaService.java

```diff
@@ -0,0 +1,205 @@
+package io.demo.bank.service;
+
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.MathContext;
+import java.security.Principal;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.hibernate.type.StringType;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.client.HttpStatusCodeException;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.demo.bank.model.AtmLocation;
+import io.demo.bank.util.Constants;
+
+@Service
+@Transactional
+public class VisaService {
+	
+	private static final Logger LOG = LoggerFactory.getLogger(VisaService.class);
+	
+	// ATM Location Service URL
+	private static String apiBaseUrl;
+
+	
+	@Autowired
+	private Environment environment;
+		
+	
+	public List<AtmLocation> searchATMLocations (String zipcode, String visaamount)	throws Exception {
```

Comment:
What is this doing here? ATM Search specific to VISA? I am confused unless this was copy pasta mistake